### PR TITLE
Adds Extract and Combine Patches Transforms

### LINKED ISSUE #113: Add transform for combining patches
For converting a large image to patches during training/inference, we can use [kornia.contrib.ExtractTensorPatches](https://kornia.readthedocs.io/en/latest/contrib.html?highlight=patches#kornia.contrib.ExtractTensorPatches). However, there is no method for converting back from patches->image on inference.

### PR DESCRIPTION
**Pull request by isaaccorley**
Added `torchgeo.transforms.ExtractPatches` and `torchgeo.transforms.CombinePatches`

Notes:
- See [example colab notebook](https://colab.research.google.com/drive/1-4fVQBhb2GlNfBDUvCBBNnzTCcc4aWY3?usp=sharing) for a visual example on a LEVIR-CD+ image and mask.
- Operates on dicts of batched tensors
- Flattens the number of patches into the batch dimension (b, c, h, w) -> (b * num_patches, c, ph, pw)
- I originally used [kornia.contrib.ExtractTensorPatches](https://kornia.readthedocs.io/en/latest/contrib.html#kornia.contrib.ExtractTensorPatches) but then realized how memory heavy it was so rolled my own using einops.

Usage:

```python
import torch
from torchgeo.transforms import ExtractPatches, CombinePatches

patch_size = (128, 128)
image_size = (1024, 1024)
image = torch.randn(1, 3, 1024, 1024)
mask = torch.ones(1, 1, 1024, 1024)
sample = dict(image=image, mask=mask)

sample_patched = ExtractPatches(patch_size)(sample)
sample_patched["image"].shape # (64, 3, 128, 128)
sample_patched["mask"].shape # (64, 1, 128, 128)

sample_combined = CombinePatches(image_size)(sample_patched)
sample_combined["image"].shape # (1, 3, 1024, 1024)
sample_combined["mask"].shape # (1, 1, 1024, 1024)

```

TODO:
- more unit tests

Closes #113

### CODE REVIEW

**Comment by adamjstewart** [comment]
My first thought when seeing this was "why do we need this function when that's the whole point of GeoSamplers?", but then I realized that this would actually work very nicely for VisionDatasets where we don't have any geospatial metadata. Can it only sample in a grid-like fashion or can it also randomly sample?

I'm not very keen on reimplementing Kornia transforms though. I would suggest first trying to submit a patch to them to make their implementation more memory efficient. If they don't accept the patch, then we can merge this PR. Our goal is not to reinvent common computer vision transforms but to use existing implementations whenever possible.

**Comment by adamjstewart** [comment]
Closing this because it's not specific to geospatial data and belongs in Kornia instead.
